### ISSUE TITLE
Document multiple return values

### ISSUE THREAD

**Issue by corykinney** [mkdocstrings/mkdocstrings#301]
**Problem**
Currently, documenting multiple return values is limited to an overall description of the `tuple`.

**Proposed Solution**
For functions with multiple return values, infer type hinting when possible and add support for position-specific variable descriptions. For the following example:

```python
def foo() -> tuple[int, float]:
    """
    Returns:
        First return value.
        Second return value.
    """
    return 1, 2.0
```

the output table should be:

```
| Type | Description               |
| int    | First return value.      |
| float | Second return value. |
```

For the general case where there is only a single description given, the output should default to the current behavior; however, type information could be improved using the return annotation (e.g. `tuple[int, float]`. 

**Discussion**
How to distinguish a multi-line description of the overall tuple from multiple single-line descriptions of individual values. Maybe something like:
```python
"""
Returns:
    1. First return value.
    2. Second return value.
"""
```

**Comment by corykinney** [comment]
@pawamoy from #300

**Comment by Saransh-cpp** [comment]
> Also, is there a desire for return parameter names, or only types and descriptions?

It would be nice to have the names too 😄

**Comment by corykinney** [comment]
> It would be nice to have the names too 😄

Yes, I agree! I just think this won't get implemented very soon, as I believe it will be transitioning from pytkdocs to a new system and most of the smaller stuff will likely have to wait until then.

**Comment by pawamoy** [comment]
This is implemented in the new handler :slightly_smiling_face: 
See https://mkdocstrings.github.io/griffe/docstrings/ for details about docstring support in Griffe.
